### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by benzrf** [celluloid/ecell#9]
Should I put a license in the repo? If so, do you already have one in mind, or would LGPL 3 be OK?

**Comment by digitalextremist** [comment]
I am tending toward `GPLv3` in this case, but `MIT` and `MPL` are not ruled out. Do you have a philosophical position to air yourself?

**Comment by benzrf** [comment]
I tend to lean more towards share-alike than total freedom. LGPL3 is still my "default" license for libraries, since I get worried about the full GPL putting people off from using it in their projects, but I don't have any specific issue with GPL if you were already planning it.

**Comment by digitalextremist** [comment]
I will take care of this while doing #8. Will discuss with contributing company and return to discussion with you on this if necessary, during same period where I do #8 at the end.

**Comment by benzrf** [comment]
OK.

**Comment by skull-squadron** [comment]
If ECell is intended to be used (e.g., production), it's a best-practice to consider {2,3}-BSD, Apache 2 || MIT license.  AGPL is Fukushima radioactive, GPL 3 is Zika, GPL 2 is the flu and LGPL is a cold. Basically all tech enterprises whom are on their game have IP lawyers whom eventually audit FOSS dependencies for licenses. 

If a freemium model, i.e., Sidekiq, MySQL, Berkeley DB, is intended, also consider multi-licensing if that helps shore-up business model "moats" from enterprise exploitation and self-promote the heck out of "keeping-the-lights on/pay the bills" support at the appropriate time, i.e. consulting, pro features, training/certification, support agreements to continue development: fix bugs (faster with pro, community with CE)/add pro/CE features.

(And, please make a killer app/use-case with ECell to develop further during a private alpha, probably by volunteering to help a Ruby shop (Heroku, Engine Yard, Github, Basecamp, etc.) optimize specific piece(s) of their infrastructure would help evolve code on the front-lines.)

**TL;DR:** Apache 2.0 👍 

See also: https://puppet.com/blog/relicensing-puppet-to-apache-2-0

**Comment by sarahzrf** [comment]
I'm gonna leave this one to @digitalextremist, given that the original code was his, and given that I don't have a ton of experience with the software industry. For what it's worth, though, I'm personally enough of a naïve idealist that I want to push share-alike. In fact, given that ECell is appropriate as a tool for building services, anything less than AGPL already seems like a compromise! (which is not to say that I don't want to compromise)
